app/controllers/reports/match_progress_controller.rb

```diff
@@ -19,7 +19,7 @@ def index
       respond_to do |format|
         format.html {}
         format.xlsx do
-          @included_sub_programs = sub_program_list.invert.slice(*report_params[:sub_programs])
+          @included_sub_programs = sub_program_list.select { |sp| sp.last.in?(report_params[:sub_programs]) }
```

Comment:
Just checking: we're confident that the sub_program_list ids are of the same type as report_params[:sub_programs] (strings / ints)?

This is totally fine as is; might be a little clearer to use filter instead of select and name the filtered vars. 
```suggestion
          @included_sub_programs = sub_program_list.filter { |_name, sp_id| sp_id.in?(report_params[:sub_programs]) }
```

Reply:
Thanks, I definitely like that syntax better.